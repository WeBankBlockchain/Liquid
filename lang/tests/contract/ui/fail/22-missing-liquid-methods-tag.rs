use liquid_lang as liquid;

#[liquid::contract]
mod noop {
    #[liquid(storage)]
    struct Noop {}

    impl Noop {
        pub fn new(&mut self) {}

        pub fn noop(&self) {}
    }
}

fn main() {}
