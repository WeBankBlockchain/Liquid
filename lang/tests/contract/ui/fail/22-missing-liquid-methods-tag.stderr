error: `impl` blocks in contract should be tagged with `#[liquid(methods)]`
  --> $DIR/22-missing-liquid-methods-tag.rs:8:5
   |
8  | /     impl Noop {
9  | |         pub fn new(&mut self) {}
10 | |
11 | |         pub fn noop(&self) {}
12 | |     }
   | |_____^
